core/src/components/input/input.tsx

```diff
@@ -338,11 +355,25 @@ export class Input implements ComponentInterface {
     this.ionInput.emit(ev as InputEvent);
   };
 
+  private onChange = () => {
+    this.emitValueChange();
+  };
+
   private onBlur = (ev: FocusEvent) => {
     this.hasFocus = false;
     this.focusChanged();
     this.emitStyle();
 
+    if (this.value != null && this.previousValue !== this.value) {
```

Comment:
The logic here is not correct. This will emit `ionChange` even if the user never commits a value.

Steps to repro:

1. Type "123" but do not click outside the input.
2. Click the "clear" button.
3. Blur the input. Observe that "ionChange" is emitted.


This is a bit tricky since there does not seem to be native browser behavior we are able to leverage for this. This also means that the `data` payload will probably be empty when `ionChange` is emitted. Perhaps this is why we decided not to emit the native event and instead emit a custom payload?

Reply:
Ok - I think I have this resolved now. Instead of tracking the `previousValue` against the last emitted `ionChange` state, we really only care about the change in the input's value from the time it was focused vs. when it is blurred. I've updated the implementation to track the value difference between these states and only emit when this happens as a result of a clear action.